packages/eslint-plugin-lit-a11y/lib/utils/HasLitHtmlImportRuleExtension.js

```diff
@@ -0,0 +1,161 @@
+      getLitHtmlSpecifiers(context).includes(getPackageName(node.source.value)));
+
+  const { specifiers = [] } = node;
+
+  return specifiers.some(isLitHtmlImport);
+}
+
+/**
+ * @param {import('estree').ImportSpecifier | import('estree').ImportDefaultSpecifier | import('estree').ImportNamespaceSpecifier} specifier
+ * @return {specifier is import('estree').ImportNamespaceSpecifier}
+ */
+function isNamespaceImport(specifier) {
+  return specifier.type === 'ImportNamespaceSpecifier';
+}
+
+/**
+ * Returns a list of lit-html tag function names
+ * @param {import('estree').ImportDeclaration & import('eslint').Rule.NodeParentExtension} node
+ * @return {string[]}
+ */
+function getLitHtmlTags(node) {
+  return node.specifiers
+    .map(specifier => {
+      switch (specifier.type) {
+        case 'ImportNamespaceSpecifier':
+          return 'html';
+        case 'ImportSpecifier':
+          if (specifier.imported.name === 'html') return specifier.local.name || 'html';
+          return '';
+        default:
+          return '';
+      }
+    })
+    .filter(Boolean);
+}
+
+/**
+ * Returns a list of lit-html tag function names
+ * @param {import('estree').ImportDeclaration & import('eslint').Rule.NodeParentExtension} node
+ * @return {string[]}
+ */
+function getLitHtmlNamespaces(node) {
+  return node.specifiers
+    .map(specifier => isNamespaceImport(specifier) && specifier.local.name)
+    .filter(Boolean);
+}
+
+const HasLitHtmlImportRuleExtension = {
+  /**
+   * @param {LitA11yRuleContext} context
+   * @return {import('eslint').Rule.RuleListener}
+   */
+  createAdditionalVisitors(context) {
+    /** if user set `litHtmlSources` to literal `false`, analyse all html-tagged literals */
+    const userExplicitlyDisabledImportValidation = context.settings.litHtmlSources === false;
+
+    /* eslint-disable no-param-reassign */
+
+    context.parserServices.shouldAnalyseHtmlTaggedLiterals = userExplicitlyDisabledImportValidation;
+
+    return {
+      ImportDeclaration(node) {
+        const shouldAnalyseNode =
+          userExplicitlyDisabledImportValidation || isLitHtmlImportDeclaration(node, context);
+
+        if (shouldAnalyseNode) {
+          context.parserServices.shouldAnalyseHtmlTaggedLiterals = true;
+
+          context.parserServices.litHtmlNamespaces = [
+            ...(context.parserServices.litHtmlNamespaces || []),
+            ...getLitHtmlNamespaces(node),
+          ];
+
+          context.parserServices.litHtmlTags = [
+            ...(context.parserServices.litHtmlTags || []),
+            ...getLitHtmlTags(node),
+          ];
+        }
+      },
+    };
+
+    /* eslint-enable no-param-reassign */
+  },
+
+  /**
+   * @param {import('eslint').Rule.RuleMetaData} meta
+   * @param {LitA11yRuleContext} context
+   */
+  reportOverrides(meta, context) {
+    const { shouldAnalyseHtmlTaggedLiterals } = context.parserServices || {};
+    return !!shouldAnalyseHtmlTaggedLiterals;
```

Comment:
What does this method do?

Reply:
https://www.npmjs.com/package/eslint-rule-extender

What were doing here iis skipping reporting

Reply:
this function can return 
- `true`: report original message
- `false`: skip reporting
- `RuleMetadata`: replace message

Reply:
gotcha 👍